AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m

```diff
@@ -292,6 +292,10 @@ + (int)deleteOldestLogsWithCount:(NSInteger)count inOpenedDatabase:(void *)db {
   NSString *selectOldestQuery = [NSString stringWithFormat:@"SELECT \"%@\" FROM \"%@\" ORDER BY \"%@\" ASC LIMIT %ld", kMSIdColumnName,
                                                            kMSLogTableName, kMSIdColumnName, (long)count];
   NSArray<NSArray *> *entries = [MSDBStorage executeSelectionQuery:selectOldestQuery inOpenedDatabase:db];
+  if (entries.count == 0) {
+    MSLogError([MSAppCenter logTag], @"There are no logs to delete.");
+    return SQLITE_ERROR;
```

Comment:
probably this is not an error case but something worth raise a flag, probably a Warning message.

Reply:
We need to break loop in this case. The warning message isn't enough

https://github.com/Microsoft/AppCenter-SDK-Apple/blob/2a8c6765e08b8e8c87af2ee011eb2fce3e3838ce/AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m#L71-L74


@jaelim-ms This can be discarded on merge

Reply:
If we can discard this PR on merge, why do we need to take this fix?

Reply:
Because the crash/task is not related to the feature and should be fixed on `develop`. Of course it doesn't make sense if feature will be merged first/right after

Reply:
Let's keep it open until we merge the feature branch since we expect the merge quite soon.